specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2023-04-01/machineLearningServices.json

```diff
@@ -4701,6 +4772,260 @@
         }
       }
     },
+    "ImageMetadata": {
+      "type": "object",
+      "description": "Returns metadata about the operating system image for this compute instance.",
+      "properties": {
+        "currentImageVersion": {
+          "type": "string",
+          "description": "Specifies the current operating system image version this compute instance is running on."
+        },
+        "latestImageVersion": {
+          "type": "string",
+          "description": "Specifies the latest available operating system image version."
+        },
+        "isLatestOsImageVersion": {
+          "type": "boolean",
+          "description": "Specifies whether this compute instance is running on the latest operating system image."
+        }
+      }
+    },
+    "CustomService": {
+      "type": "object",
+      "description": "Specifies the custom service configuration",
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "Name of the Custom Service"
+        },
+        "image": {
+          "$ref": "#/definitions/Image",
+          "description": "Describes the Image Specifications"
+        },
+        "environmentVariables": {
+          "type": "object",
+          "description": "Environment Variable for the container",
+          "additionalProperties": {
+            "$ref": "#/definitions/EnvironmentVariable"
+          }
+        },
+        "docker": {
+          "description": "Describes the docker settings for the image",
+          "$ref": "#/definitions/Docker",
+          "x-nullable": true
+        },
```

Comment:
Are these 3 properties being used and how?

environmentVariables
docker
volumes

Reply:
those are docker specific values. docker gets merged into docker config. advanced stuff for advanced users.

valumes is how you mount volumes (a docker thing)

environmentVariables set environment variables

Reply:
Thank you.